File: app/src/main/java/ru/illarionovroman/yandexmobilizationhomework/ui/activity/AboutActivity.java
package ru.illarionovroman.yandexmobilizationhomework.ui.activity;

import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.Toast;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.illarionovroman.yandexmobilizationhomework.R;


/**
 * Activity with About content.
 */
public class AboutActivity extends AppCompatActivity {

    @BindView(R.id.ivLogo)
    ImageView mIvLogo;

    private Toast mToast;
    private int mClickCount = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_about);
        ButterKnife.bind(this);
        initializeActivity();
    }

    private void initializeActivity() {
        initializeActionBar();
        setLogoClickListener();
    }

    private void initializeActionBar() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbarAbout);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setTitle(R.string.about_title);
        }
    }

    private void setLogoClickListener() {
        int secretCount = 10;
        mIvLogo.setOnClickListener(view -> {
            mClickCount++;
            if (2 < mClickCount && mClickCount < secretCount) {
                showToast(String.valueOf(secretCount - mClickCount) + " more to go...");
            } else if (mClickCount >= secretCount) {
                showToast(getString(R.string.easter_egg));
            }
        });
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    /**
     * Cancel currently displayed toast before showing the new one
     */
    private void showToast(String textToShow) {
        if (mToast != null) {
            mToast.cancel();
        }
        mToast = Toast.makeText(this, textToShow, Toast.LENGTH_SHORT);
        mToast.show();
    }
}

File: app/src/main/java/ru/illarionovroman/yandexmobilizationhomework/ui/activity/AboutActivity.kt
package ru.illarionovroman.yandexmobilizationhomework.ui.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.widget.ImageView
import android.widget.Toast
import ru.illarionovroman.yandexmobilizationhomework.R

class AboutActivity : AppCompatActivity() {

    var mIvLogo: ImageView? = null

    var mToast: Toast? = null
    var mClickCount: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)
        bindViews()
        initializeActivity()
    }

    private fun bindViews() {
        mIvLogo = findViewById(R.id.ivLogo) as ImageView
    }

    private fun initializeActivity() {
        initializeActionBar()
        setLogoClickListener()
    }

    private fun initializeActionBar() {
        val toolbar: Toolbar? = findViewById(R.id.toolbarAbout) as Toolbar
        setSupportActionBar(toolbar)
        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            setTitle(R.string.about_title)
        }
    }

    private fun setLogoClickListener() {
        val secretCount: Int = 10
        mIvLogo?.setOnClickListener {
            mClickCount++
            if (mClickCount in 3..(secretCount - 1)) {
                val clicksLeft: Int = secretCount - mClickCount
                showToast(clicksLeft.toString() + " more to go...")
            } else if (mClickCount >= secretCount) {
                showToast(getString(R.string.easter_egg))
            }
        }
    }

    private fun showToast(textToShow: String) {
        mToast?.cancel()
        mToast = Toast.makeText(this, textToShow, Toast.LENGTH_SHORT)
        mToast!!.show()
    }
}
